### ISSUE TITLE
Cannot run with `ensure_undelete` in Terraform Cloud

### ISSUE THREAD

**Issue by djbingham** [terraform-google-modules/terraform-google-endpoints-dns#17]
### TL;DR

This module cannot run without disabling the `ensure_undelete` option in environments where gcloud is not pre-installed, such as Terraform Cloud.

### Expected behavior

This module should be able to run with any valid configuration in Terraform Cloud.

### Observed behavior

If `ensure_undelete` is enabled (default setting), execution fails due to a "gcloud: command not found" error in `cloudep_prep.sh`.

For example:
```
Error: External Program Execution Failed
with module.gke.module.cluster_dns.data.external.module-cloudep-dns-prep[0]
on .terraform/modules/gke.cluster_dns/main.tf line 50, in data "external" "module-cloudep-dns-prep":
  program = ["${path.module}/scripts/cloudep_prep.sh"]
The data source received an unexpected error while attempting to execute the program.

Program: .terraform/modules/gke.cluster_dns/scripts/cloudep_prep.sh
Error Message: ++ command -v jq
+ JQ=/usr/bin/jq
+ [[ -z /usr/bin/jq ]]
++ /usr/bin/jq -r '@sh "ENDPOINT=\(.endpoint) PROJECT=\(.project)"'
+ eval 'ENDPOINT='\''cluster.endpoints.impactful-test-1e3b.cloud.goog'\'' PROJECT='\''impactful-test-1e3b'\'''
++ ENDPOINT=cluster.endpoints.impactful-test-1e3b.cloud.goog
++ PROJECT=impactful-test-1e3b
++ mktemp -d
+ TMP_DIR=/tmp/tmp.aBRIKhhyKZ
+ trap cleanup EXIT
+ [[ ! -z x ]]
+ [[ ! -z '' ]]
+ log INFO 'Enabling Service Management API'
+ level=INFO
+ msg='Enabling Service Management API'
+ echo ': Enabling Service Management API'
: Enabling Service Management API
+ gcloud services enable servicemanagement.googleapis.com serviceusage.googleapis.com --project impactful-test-1e3b
.terraform/modules/gke.cluster_dns/scripts/cloudep_prep.sh: line 41: gcloud: command not found
+ cleanup
+ rm -rf /tmp/tmp.aBRIKhhyKZ

State: exit status 127
```

### Terraform Configuration

```hcl
The relevant part of my configuration is (within a parent module named `gke`):

module "cluster_dns" {
  source      = "terraform-google-modules/endpoints-dns/google"
  version     = "3.0.1"

  external_ip = "${module.cluster.endpoint}"
  name        = "cluster"
  project     = "${var.gcp_project_id}"
}
```
```


### Terraform Version

```sh
1.1.9
```

### Additional information

I believe this issue is due to the use of `gcloud` commands in `cloudep_prep.sh`.

Several other modules have resolved similar issues by using the [terraform-google-gcloud](https://github.com/terraform-google-modules/terraform-google-gcloud) module to execute scripts containing `gcloud` commands. I propose using the same approach here - execute `cloudep_prep.sh` using the gcloud module rather than an external data source. This would be in line with the [Terraform documentation](https://registry.terraform.io/providers/hashicorp/external/latest/docs/data-sources/data_source), which warns against using external data sources except for "exceptional situations where a first-class Terraform provider is not more appropriate".

I intend to have a go at this myself and submit a PR if I can get it working.
